### PR TITLE
ENG-95 Override Taurus Engine's REST API Key in Taurus pipelines

### PR DESCRIPTION
**Pull request by vitaly-krugl**
ENG-95 Impelement taurus-set-api-key console script for overriding Taurus Engine's REST API Key.

ENG-95 Updated Taurus pipeline scripts to override Taurus Engine's REST API Key using the new tool.

### CODE REVIEW

File: taurus/conf/application.conf [hunk 1]
```diff
@@ -40,7 +40,6 @@ plaintext_port = 2003
 queue_name = taurus.metric.custom.data
 
 [security]
-# TAUR-1065 Remove this from the conf file and replace with something else (env var??)
```
Comment (vitaly-krugl) [inline taurus/conf/application.conf]: TAUR-1065 was closed

**Comment by vitaly-krugl** [comment]
Merging as hotfix to expedite resolution of failure on staging.

Please review

CC @scottpurdy, @oxtopus

File: taurus/taurus/engine/set_api_key.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+g_log = logging.getLogger(__name__)
+
+
+
+def main():
+  logging_support.LoggingSupport().initTool()
+
+
+  parser = argparse.ArgumentParser(description=__main__.__doc__)
```
Comment (oxtopus) [inline taurus/taurus/engine/set_api_key.py]: Better to use `__doc__` alone as opposed to `__main__.__doc__`, as `__main__` may change, depending on how the code is run, and `__main__.__doc__` will refer to the docstring of the `__main__` in which this code is executed (and not this module).

File: taurus/taurus/engine/set_api_key.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+"""Applies the given Taurus REST API key as override for application.conf."""
+
+
+import logging
+import __main__
```
Comment (oxtopus) [inline taurus/taurus/engine/set_api_key.py]: See my other comment about `__main__`, you shouldn't need to import `__main__`.

File: taurus/taurus/engine/set_api_key.py (hunk 2 above)
Comment (vitaly-krugl) [inline taurus/taurus/engine/set_api_key.py]: Good to know. Fixed.

File: taurus/taurus/engine/set_api_key.py (hunk 3 above)
Comment (vitaly-krugl) [inline taurus/taurus/engine/set_api_key.py]: removing import of __main__

**Comment by vitaly-krugl** [comment]
@oxtopus: thanks, I addressed your feedback (including from email) via PR #463

**Comment by vitaly-krugl** [comment]
@oxtopus, are you done with this code review, and are the fixes in #463 sufficient?

**Comment by oxtopus** [comment]
I'm done.

File: taurus/taurus/engine/set_api_key.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+  if not args.apikey:
+    msg = "Missing or empty api key"
+    g_log.error(msg)
+    parser.error(msg)
+
+
+  conf = taurus.engine.config
+
+  assert conf.has_section("security"), (
```
Comment (scottpurdy) [inline taurus/taurus/engine/set_api_key.py]: This seems like an important check to always perform so I'd change this to an explicit if/raise. The `assert` statement is only checked when `__debug__` is set to true. Same for below.

Comment (vitaly-krugl) [inline taurus/taurus/engine/set_api_key.py]: @scottpurdy, the implementation is based on the assumption that the conf file must contain that section + option. The asserts are here to verify this assumption. If you thing that this intended scenario warrants an explicit check, I can get it in.

Comment (scottpurdy) [inline taurus/taurus/engine/set_api_key.py]: Oh I see, this is the configuration that ships with Taurus (possibly with some overrides), not some user-defined configuration? Fine to leave as an assert then.

Comment (vitaly-krugl) [inline taurus/taurus/engine/set_api_key.py]: @scottpurdy, that's correct.

**Pull request merged**